frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java

```diff
@@ -139,4 +141,46 @@ public void extract(String archive, String destinationDirectory) throws ArchiveE
                     + "'", e);
         }
     }
+
+    /**
+     * Do multiple file system checks that should enable the plugin to work on any file system
+     * whether or not it's case sensitive or not.
+     *
+     * @param destPath
+     * @param destDir
+     * @return
+     */
+    private boolean startsWithPath(String destPath, String destDir) {
+        if (destDir.startsWith(destDir)) {
+            return true;
+        } else if (destDir.length() > destPath.length()) {
+            return false;
+        } else {
+            /*
+             * The first check should handle case-sensitive file systems. We need this
+             * in order to weed out case-sensitive file systems with a non-existent path
+             * that slipped through the first test.
+             */
+            if (new File(destPath).exists() && !(new File(destPath.toLowerCase()).exists())) {
+                return false;
+            }
+
+            boolean retVal = true;
```

Comment:
Why is this stuff going on? Isn't it easier with `destPath.toLowerCase().startsWith(destDir.toLowerCase())`?